src/main/java/com/course/fleet/management/api/controller/UserController.java

```diff
@@ -0,0 +1,72 @@
+package com.course.fleet.management.api.controller;
+
+import com.course.fleet.management.api.entity.User;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import net.datafaker.Faker;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+  private static final List<User> users;
+
+  static {
+    Faker faker = new Faker();
+    users =
+        IntStream.range(0, 10)
+            .mapToObj(
+                i -> {
+                  String firstName = faker.name().firstName();
+                  String lastName = faker.name().lastName();
+                  return new User(
+                      (long) i,
+                      firstName + " " + lastName,
+                      firstName + "." + lastName + "@mail.com");
+                })
+            .collect(Collectors.toList());
+  }
+
+  @GetMapping
+  public ResponseEntity<List<User>> getUsers() {
+    return ResponseEntity.ok(users);
+  }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<Optional<User>> getUser(@PathVariable long id) {
```

Comment:
Hay que retornar al cliente solo el User. De nuestro lado nosotros nos encargaremos de manejar siempre los objetos opcionales.

```suggestion
  public ResponseEntity<User> getUser(@PathVariable long id) {
```